### ISSUE TITLE
Make handling asynchronous auth easier to handle

### ISSUE THREAD

**Issue by weissi** [grpc/grpc-swift#1618]
Currently, auth is handled with `Client/ServerInterceptor`s and that of course works. But this has a number of pitfalls:

- It requires you to call `send`/`receive` on the correct `EventLoop` (that could actually be fixed)
- It requires you to call the `send`/`receive`s in the order you got called. That means for async auth you will need to manually buffer! (see for example this user's illegal state issue: https://stackoverflow.com/questions/76534719/eventloopfuture-causing-invalid-state-unable-to-write-message/76535053#76535053 as well as https://github.com/grpc/grpc-swift/issues/1181#issuecomment-842116341)

Basically it's way too hard to get this right for something that many/most(?) people will need.
